scripts/create-hsm-wallet.js

```diff
@@ -44,12 +44,15 @@ async function createXprivKey(keyName) {
 }
 
 /**
- * Creates a promise to delay for a given number of milisseconds
- * @param {number} [ms=1000]
+ * Creates a promise to delay for a given number of milliseconds
+ * @param {number} ms Number of milliseconds to delay. Allows zero.
  * @returns {Promise<void>}
  */
 async function delay(ms) {
-  return new Promise(resolve => { setTimeout(resolve, ms || 1000); });
+  if (ms === null || ms === undefined || isNaN(ms)) {
```

Comment:
Maybe just add a comment here that you need to do all checks but javascript has some weird behaviors, e.g., `isNaN(null) = false`.

Reply:
Replaced this check with the lodash `isNumber()` method for increased legibility. Also added a validation for non-negative numbers:  `>= 0`